client/clientservice/src/request_service.cpp

```diff
@@ -44,30 +44,37 @@ Status RequestServiceImpl::Send(ServerContext* context, const Request* proto_req
 
   auto callback = [&](cc::SendResult&& send_result) {
     if (not std::holds_alternative<bft::client::Reply>(send_result)) {
-      LOG_INFO(logger_, "Send returned error");
       switch (std::get<uint32_t>(send_result)) {
         case (static_cast<uint32_t>(bftEngine::OperationResult::INVALID_REQUEST)):
+          LOG_INFO(logger_, "Request failed with INVALID_ARGUMENT error for cid=" << req_config.correlation_id);
           status.set_value(grpc::Status(grpc::StatusCode::INVALID_ARGUMENT, "Invalid argument"));
           break;
         case (static_cast<uint32_t>(bftEngine::OperationResult::NOT_READY)):
+          LOG_INFO(logger_, "Request failed with INVALID_ARGUMENT error for cid=" << req_config.correlation_id);
```

Comment:
An incorrect error is written to the log. It should be NOT_READY here. Please fix all other logs, too.

Reply:
sure, changes were not added. will add them . thanks